client/src/app/components/CreateProject/Options/OptionsContainer.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import Option from "./Option";
+
+class OptionsContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { countActive: 0, isMax: false };
+  }
+
+  componentWillUpdate({ reset }) {
+    if (reset !== this.props.reset) {
+      if (!this.props.reset) {
+        this.setState(() => ({ isMax: false, countActive: 0 }));
+      }
+    }
+  }
+
+  checkMax(num) {
+    this.setState(
+      prevState => ({ countActive: prevState.countActive + num }),
+      () => {
+        if (this.state.countActive === this.props.maxAllowed) {
+          this.setState({ isMax: true });
+        } else {
+          this.setState({ isMax: false });
+        }
+      }
+    );
+  }
+
+  render() {
+    const { options, onUpdateSelection, isDefault, reset } = this.props;
+    const { isMax } = this.state;
```

Comment:
This is great! You could even leave an empty row after variable declaration  for better visual separation